doc/030_Advanced_Topics.md

```diff
@@ -17,6 +17,20 @@ The container will be re-created on the next call of `./exec.bash`.
 
 When you update the image by pulling from registry or rebuilding the devel image, the containers are also regenerated.
 
+### Keeping the Fork in Sync with Upstream Changes
+
+Normally it is not required to keep the forked repo in sync with the upstream changes.
+Everything should be fine, as long as the forked repo version corresponds to the version that was used to create the devel image that is used.
+
+A common case for when a sync actually is (or may be) required, is the creation of a new devel image.
+The scripts' version should match that with which the upstream-provided base image was created.
+Recent versions of the scripts implement a version check that compares the versions of scripts and base image for this purpose.
+A warning is printed if the versions no longer match.
+
+In this case you should merge the upstream changes with your setup.
+
+In case you want to build a new devel based on a new version of the base image you might need to activate the setting `DOCKER_REGISTRY_AUTOPULL` to make sure a new version is actually pulled from the registry (if available).
+
```

Comment:
On the other hand, if `DOCKER_REGISTRY_AUTOPULL` is false, you'll re-use the base image used for the initial build of the devel image (if you still have it available). Also, you can build your own local base images using the script versions in your fork.